### DISCUSSION TITLE
[Feature] Allow refreshing facial recognition guesses after user corrects an incorrect face

### DISCUSSION THREAD

**Discussion by Germs2004** [immich-app/immich#12933] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

If Immich assigns a face in a photo to the wrong person, and I edit that face to the correct person, I'd like a way to tell Immich to rescan all the faces and consider that new information so it will automatically correct any other mistakes that are similar to this one.

For example, I may have 1000 photos of John and 1000 of Sally, but 20 photos of Sally are incorrectly marked as John, so when I edit one of those mistakes from John to Sally, Immich should fix the other 19 automatically.

This feature request requires #12931 as a prerequisite so it won't overwrite a face that the user has already confirmed is correct.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by TheMClaeys** [comment]
That would be usefull, I have some mixups, too many to fix manually.
I saw a question on Reddit about the same feature.

**Comment by ty-han** [comment]
Absolutely necessary.  I thought this is what the "refresh" feature does. If not then what is refresh?

**Comment by sikala** [comment]
this would be fantastic. I have family photos were all the kids/cousins etc were identified as the same person. I would like to fix a few then have it refresh all the the others to be correct. Surely even immich should know that 4 people in the same picture are not the same person.
